services/provider/server/server.go

```diff
@@ -500,42 +516,48 @@ func (s *OCSProviderServer) appendClientProfileExternalResources(
 	// same name
 	profileMap := make(map[string]*csiopv1a1.ClientProfileSpec)
 
-	rbdClientProfileName := consumerConfig.GetRbdClientProfileName()
-	rbdClientProfile := profileMap[rbdClientProfileName]
-	if rbdClientProfile == nil {
-		rbdClientProfile = &csiopv1a1.ClientProfileSpec{
-			CephConnectionRef: corev1.LocalObjectReference{Name: consumer.Status.Client.Name},
-		}
-		profileMap[rbdClientProfileName] = rbdClientProfile
+	if availableServices.Rbd {
```

Comment:
In this context you don't need the available services map, you already decided what is available based on the keys in the configmap